### PR TITLE
Make unmanaged clients a first class citizen

### PR DESCRIPTION
**Pull request by abrooksv**
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/tst/software/aws/toolkits/jetbrains/core/MockClientManager.kt [hunk 1]
```diff
@@ -24,17 +25,18 @@ import kotlin.reflect.KClass
 class MockClientManager : AwsClientManager() {
     private data class Key(
         val clazz: KClass<out SdkClient>,
-        val region: AwsRegion? = null,
-        val credProviderId: String? = null
+        val region: String? = null,
```
Comment (rli) [inline jetbrains-core/tst/software/aws/toolkits/jetbrains/core/MockClientManager.kt:28]: rename to `regionId`?

File: core/src/software/aws/toolkits/core/ToolkitClientManager.kt [hunk 2]
```diff
@@ -55,89 +52,98 @@ abstract class ToolkitClientManager {
...
     }
 
+    private fun <T : SdkClient> createNewClient(sdkClass: KClass<T>, connection: ConnectionSettings): T = constructAwsClient(
+        sdkClass = sdkClass,
+        credProvider = connection.credentials,
+        region = Region.of(connection.region.id),
+    )
+
+    /**
+     * Constructs a new low-level AWS client whose lifecycle is **NOT** managed centrally. Caller is responsible for shutting down the client
```
Comment (rli) [inline core/src/software/aws/toolkits/core/ToolkitClientManager.kt:70]: are you envisioning this only being used for short-lived clients? what would be the best way to deal with long-lived clients

File: jetbrains-core/tst/software/aws/toolkits/jetbrains/core/MockClientManager.kt (hunk 1 above)
Comment (abrooksv) [reply jetbrains-core/tst/software/aws/toolkits/jetbrains/core/MockClientManager.kt:28]: Good call

File: core/src/software/aws/toolkits/core/ToolkitClientManager.kt (hunk 2 above)
Comment (abrooksv) [reply core/src/software/aws/toolkits/core/ToolkitClientManager.kt:70]: Long lived too, like Telemetry uses it and thats the life of the toolkit. It just means that the callee is responsible for calling close on the client, not the toolkit on dispose

**Review by rli — APPROVED** [review]

**Pull request merged**